app/javascript/mastodon/features/ui/components/media_modal.js

```diff
@@ -67,10 +72,17 @@ export default class MediaModal extends ImmutablePureComponent {
     const { media, intl, onClose } = this.props;
 
     const index = this.getIndex();
+    const pagination = [];
 
     const leftNav  = media.size > 1 && <button tabIndex='0' className='modal-container__nav modal-container__nav--left' onClick={this.handlePrevClick} aria-label={intl.formatMessage(messages.previous)}><i className='fa fa-fw fa-chevron-left' /></button>;
     const rightNav = media.size > 1 && <button tabIndex='0' className='modal-container__nav  modal-container__nav--right' onClick={this.handleNextClick} aria-label={intl.formatMessage(messages.next)}><i className='fa fa-fw fa-chevron-right' /></button>;
 
+    if (media.size > 1) {
+      media.forEach((item, i) => {
+        pagination.push(<li className='media-modal__list' key={i}><button tabIndex='0' className={`media-modal__button${i === index ? ' active' : ''}`} onClick={this.handleChangeIndex} data-index={i}>{i + 1}</button></li>);
```

Comment:
`.active` is still nested here. it should be `${i === index ? ' media-modal__button--active' : ''}`

furthermore, keeping the classname stuff inline is pretty hard to read. I would use `classes()` and pull out the entire class list to an intermediate variable.